_source/_posts/2023-07-27-spring-graphql-react.md

```diff
@@ -0,0 +1,1794 @@
+
+```
+✪ Welcome to the Auth0 CLI 🎊
+
+If you don't have an account, please create one here: https://auth0.com/signup.
+
+Your device confirmation code is: KGFL-LNVB
+
+ ▸    Press Enter to open the browser to log in or ^C to quit...
+
+Waiting for the login to complete in the browser... ⣻Opening in existing browser session.
+Waiting for the login to complete in the browser... done
+
+ ▸    Successfully logged in.
+ ▸    Tenant: dev-avup2laz.us.auth0.com
+```
+
+The next step is to create a client app, which you can do in one command:
+
+```shell
+auth0 apps create \
+  --name "GraphQL Server" \
+  --description "Spring Boot GraphQL Resource Server" \
+  --type regular \
+  --callbacks http://localhost:8080/login/oauth2/code/okta \
+  --logout-urls http://localhost:8080 \
+  --reveal-secrets
+```
+
+Once the app is created, you will see the OIDC app's configuration:
+
+```
+=== dev-avup2laz.us.auth0.com application created
+
+  CLIENT ID            ***
+  NAME                 GraphQL Server
+  DESCRIPTION          Spring Boot GraphQL Resource Server
+  TYPE                 Regular Web Application
+  CLIENT SECRET        ***
+  CALLBACKS            http://localhost:8080/login/oauth2/code/okta
+  ALLOWED LOGOUT URLS  http://localhost:8080
+  ALLOWED ORIGINS
+  ALLOWED WEB ORIGINS
+  TOKEN ENDPOINT AUTH
+  GRANTS               implicit, authorization_code, refresh_token, client_credentials
+
+ ▸    Quickstarts: https://auth0.com/docs/quickstart/webapp
+ ▸    Hint: Emulate this app's login flow by running `auth0 test login ***`
+ ▸    Hint: Consider running `auth0 quickstarts download ***`
+```
+
+Add the `okta-spring-boot-starter` dependency to the `build.gradle` file:
+
+__build.gradle__
+```groovy
+dependencies {
+    ...
+    implementation 'com.okta.spring:okta-spring-boot-starter:3.0.5'
+    ...
+}
+```
+
+Set the clientId, issuer, and audience for OAuth2 in the `application.properties` file:
+
+__application.properties__
+```properties
+okta.oauth2.issuer=https://{yourAuth0Domain}/
+okta.oauth2.client-id={clientId}
+okta.oauth2.audience=https://{yourAuth0Domain}/api/v2/
+```
+
+Add the client secret to the `.env` file:
+
+__.env__
+```shell
+export OKTA_OAUTH2_CLIENT_SECRET={clientSecret}
+```
+
+Add the following factory method to the class `SpringBootGraphQlApiConfig`, for requiring a bearer token for all requests:
+
+__SpringBootGraphQlApiConfig.java__
+```java
+    ...
+    @Bean
+    public SecurityFilterChain configure(HttpSecurity http) throws Exception {
+        http.oauth2ResourceServer(oauth2ResourceServer -> oauth2ResourceServer.jwt(withDefaults()));
+        return http.build();
+    }
+    ...
+```
+
+> __NOTE:__
+> The Okta Spring Boot starter provides the security auto-configuration out of the box, and the resource server configuration would not be necessary. For some reason, the Spring for GraphQL cors allowed origins configuration does not take effect without the customization above.
```

Comment:
```suggestion
> The Okta Spring Boot starter provides the security auto-configuration out of the box, and the resource server configuration should not be necessary. For some reason, the Spring for GraphQL cors allowed origins configuration does not take effect without the customization above.
```

You are correct. I was able to verify this. Can you please enter an issue in the Okta Spring Boot starter project?

Reply:
I will entre the issue and link it here.